docs/overview.rst

```diff
@@ -0,0 +1,83 @@
+.. _concepts:
+
+Overview
+========
+The following is a walk-through of the important concepts in the access control
+system.
+
+Users & roles
+-------------
+First we have **subjects** which can be granted access to a protected resource.
+
+- **User**: Represents an authenticated account in the system.
+- **Role**: Represents a job function. Roles are created by e.g. system
+  administrators and defined by a name. Users can be assigned zero or more
+  roles.
+- **System role**: Represents special roles that are created and defined by the
+  system and automatically assigned to users (i.e. system roles cannot be
+  created and defined by system administrators).
+
+Permissions and needs
+---------------------
+Second, we have two entities to describe access control:
+
+- **Need**: A need represents the smallest level of access control. It is very
+  generic and can express statements such as *"has admin role"* and
+  *"has read access to record 42"*.
+- **Permission**: Represents a set of required *needs*, any of which should
+  be fulfilled to access a resouce. E.g. a permission can combine the two
+  statements above into **"has admin role or has read access to record 42"**.
+
+The concept of a *need* can be somewhat hard to grasp at first, so let's
+dive in to describe how a need is used. Essentially *needs* are used to express
+a) what a permission require and b) what a user provides, i.e.:
+
+- A permission **requires** a set of needs.
+- A user **provides** a set of needs.
+
+Thus, checking if a user can access a resource protected by a permission
+amounts to checking for a non-empty **intersection** between the above sets.
+
+Types of needs
+--------------
+Needs can have different types. For instance the statement *"has admin role"*
+can be expressed as a *role need type* with the argument ``admin``. This means
+that a permission can require the admin role need and that a user can provide
+the admin role need. Some basic need types include:
+
+- **User need**: Can be used to restrict access to a specific named user.
+- **Role need**: Can be used to restrict access to a specific named role.
+
```

Comment:
I'm not entirely sure we should speak about the user and role need... it is currently not used in invenio (or rarely), and it complicates a lot the understanding of this section